src/raft/raft.rs

```diff
@@ -182,9 +188,12 @@ pub struct Raft<T: Storage> {
     pub max_inflight: usize,
     pub max_msg_size: u64,
     pub prs: FlatMap<u64, Progress>,
+    pub learner_prs: FlatMap<u64, Progress>,
```

Comment:
It seems `learner_prs` and `prs` are almost always handled together, why not put them into a struct?

Reply:
@xiang90 suggests separating them. Let's keep the same with etcd.

Reply:
I'm not suggesting to unify them, but use a struct `Prs { prs: map, leaner_prs: map }` to manage them all. It's easier to deal with ownership in that way.